test/integration/targets/unarchive/tasks/main.yml

```diff
@@ -34,6 +34,26 @@
   pkgng: name=zip,unzip state=present
   when: ansible_pkg_mgr  ==  'pkgng'
 
+- name: check that zip is installed
+  shell: zip --help
+  register: zip_is_installed
+  ignore_errors: yes
+
+- name: check that unzip has option -P (pass password from cli)
+  shell: unzip -P
+  register: unzip_has_modifier_P
+  ignore_errors: yes
+
+- name: check that pexpect is installed
+  shell: python -c "import pexpect"
```

Comment:
Use the `pip` module to install `pexpect` instead of detecting it, otherwise the tests will be skipped in CI.